go/vt/srvtopo/resilient_server.go

```diff
@@ -79,17 +79,11 @@ type ResilientServer struct {
 
 // NewResilientServer creates a new ResilientServer
 // based on the provided topo.Server.
-func NewResilientServer(ctx context.Context, base *topo.Server, counterPrefix string) *ResilientServer {
+func NewResilientServer(ctx context.Context, base *topo.Server, counts *stats.CountersWithSingleLabel) *ResilientServer {
```

Comment:
@deepthi I don't think we should check it. That's not something we do anywhere. By that argument we could add manual `nil` checks everywhere really and code would get very noisy. I think we should not do that, because it becomes very arbitrary where we would check that or not.

Reply:
By this argument we should also check it for example in `NewSrvKeyspaceWatcher` & `NewSrvVSchemaWatcher`. But why limit it to that? We pass in `*topo.Server`, why not check that too? This really spirals into checking everything and I don't think that's a good idea. And this argument is not really different to any of the other ones we have (let alone in any other random function in the code base).

Reply:
Alright I'll buy it.